src/Game.js

```diff
@@ -0,0 +1,253 @@
+import { Grid } from '@mui/material';
+import React, { createRef, useState } from 'react';
+import { Board } from './Board';
+import { BoardSize } from './BoardSize';
+import { DisplayPlayerScore } from './DisplayPlayerScore';
+import { getWinner } from './gameLogic';
+import { LoginForm } from './LoginForm';
+import { MoveBall } from './MoveBall';
+import { Winner } from './Winner';
+
+const Game = () => {
+
+    const [boardSize, setBoardSize] = useState({ row: 8, col: 8 });
+    const [nInARow, setNInARow] = useState(4);
+
+    const [playerOne, setPlayerOne] = useState({ name: 'playerOne', score: 0 });
+    const [playerTwo, setPlayerTwo] = useState({ name: 'playerTwo', score: 0 });
+
+    const [gameStarted, setGameStarted] = useState(false);
+    const [restartGame, setRestartGame] = useState('');
+
+    const [viewLoginForm, setViewLoginForm] = useState(false);
+    const [viewBoard, setViewBoard] = useState(false);
+
+    const [winner, setWinner] = useState('');
+    const [winnerBall, setWinnerBall] = useState('');
+
+    const playersProps = { playerOne, setPlayerOne, playerTwo, setPlayerTwo };
+
+    const moveBallRef = createRef();
+
+    const isValidUsername = (name, value) => {
+        const isValidName = value.length >= 3;
+
+        switch (name) {
+            case 'playerOne':
+            case 'playerTwo':
+                return isValidName;
+            default:
+                break;
+        };
+    };
+
+    const isValidNumber = (name, value) => {
+        const isValid = Number(value) >= 4;
+        const isValidNInARow = Number(value) >= 2;
+
+        switch (name) {
+            case 'rows':
+            case 'columns':
+                return isValid;
+            case 'nInARows':
+                return isValidNInARow;
+            default:
+                break;
+        };
+    };
+
+    const handleName = (event) => {
```

Comment:
handleNameChange, handleBoardSizeChange - names should indicate that a change has been made
onFormLogin, onGameStart - I'm suggesting new names since these are used for initialization